src/mkdocs_bibtex/utils.py

```diff
@@ -145,8 +214,29 @@ def insert_citation_keys(citation_quads, markdown):
 
     grouped_quads = [list(g) for _, g in groupby(citation_quads, key=lambda x: x[0])]
     for quad_group in grouped_quads:
-        full_citation = quad_group[0][0]  # the first key in the whole citation
+        full_citation = quad_group[0][0]  # the full citation block
         replacement_citaton = "".join(["[^{}]".format(quad[2]) for quad in quad_group])
+
+        # if cite_inline is true, convert full_citation with pandoc and add to replacement_citaton
+        if csl and bib:
+            try:
+                # Verify that the pandoc installation is newer than 2.11
+                pandoc_version = pypandoc.get_pandoc_version()
+                pandoc_version_tuple = tuple(int(ver) for ver in pandoc_version.split("."))
+                if pandoc_version_tuple <= (2, 11):
+                    raise Exception(f"Your version of pandoc (v{pandoc_version}) is incompatible with "
+                                    "the cite_inline feature.")
+
+                inline_citation = _convert_pandoc_citekey(bib, csl, full_citation)
+                replacement_citaton = f" {inline_citation}{replacement_citaton}"
+
+                # Make sure inline citations doesn't get an extra whitespace by
+                # replacing it with whitespace added first
+                markdown = markdown.replace(f" {full_citation}", replacement_citaton)
+
+            except Exception as e:
```

Comment:
What kind of errors are you trying to catch here? Catching blanket `Exception` is bad because even ctrl+c will get caught and ignored here.

Reply:
I originally intended to do a RuntimeError, but changed it print a warning and fallback to the old method instead. In hindsight just doing the RuntimeError makes more sense. Give me some minutes and I'll resolve this